presto-main/src/main/java/com/facebook/presto/execution/FutureStateChange.java

```diff
@@ -46,14 +49,30 @@
 
     public void complete(T newState)
     {
+        fireStateChange(newState, null);
```

Comment:
I think you can use `MoreExecutors.directExecutor` (with static import) instead of null, which makes the fire code simper